url20b0f880a5_jgraph_legacy_jgraph5_tgz-pJ8-org_jgraph_example_HelloWorldJ8/src/org/jgraph/graph/VertexView.java

```diff
@@ -0,0 +1,512 @@
+			if (!firstOverlayInvocation) {
+				if (cachedBounds != null) {
+					g.setColor(Color.black);
+					Rectangle2D tmp = graph.toScreen((Rectangle2D) cachedBounds
+							.clone());
+					g.drawRect((int) tmp.getX(), (int) tmp.getY(), (int) tmp
+							.getWidth() - 2, (int) tmp.getHeight() - 2);
+				} else if (!initialBounds.equals(vertex.getBounds())) {
+					Graphics2D g2 = (Graphics2D) g;
+					AffineTransform oldTransform = g2.getTransform();
+					g2.scale(graph.getScale(), graph.getScale());
+					graph.getUI()
+							.paintCell(g, vertex, vertex.getBounds(), true);
+					if (contextViews != null) {
+						for (int i = 0; i < contextViews.length; i++) {
+							graph.getUI().paintCell(g, contextViews[i],
+									contextViews[i].getBounds(), true);
+						}
+					}
+					if (!graph.isPortsScaled())
+						g2.setTransform(oldTransform);
+					if (portViews != null && graph.isPortsVisible())
+						graph.getUI().paintPorts(g, portViews);
+					g2.setTransform(oldTransform);
+				}
+			}
+			firstOverlayInvocation = false;
+		}
+
+		/**
+		 * Invoked when the mouse pointer has been moved on a component (with no
+		 * buttons down).
+		 */
+		public void mouseMoved(MouseEvent event) {
+			if (vertex != null) {
+				for (int i = 0; i < r.length; i++) {
+					if (r[i].contains(event.getPoint())) {
+						graph.setCursor(new Cursor(cursors[i]));
+						event.consume();
+						return;
+					}
+				}
+			}
+		}
+
+		/** Process mouse pressed event. */
+		public void mousePressed(MouseEvent event) {
+			if (!graph.isSizeable())
+				return;
+			for (int i = 0; i < r.length; i++) {
+				if (r[i].contains(event.getPoint()) && cursors[i] != 0) {
+					Set set = new HashSet();
+					set.add(vertex.getCell());
+					contextViews = context.createTemporaryContextViews(set);
+					Object[] all = AbstractCellView
+							.getDescendantViews(new CellView[] { vertex });
+					if (all.length >= org.jgraph.plaf.basic.BasicGraphUI.MAXHANDLES)
+						cachedBounds = (Rectangle2D) initialBounds.clone();
+					event.consume();
+					index = i;
+					return;
+				}
+			}
+		}
+
+		/** Process mouse dragged event. */
+		public void mouseDragged(MouseEvent event) {
+			if (firstDrag && graph.isDoubleBuffered() && cachedBounds == null) {
+				initOffscreen();
+				firstDrag = false;
+			}
+			Rectangle2D dirty = null;
+			Graphics g = (offgraphics != null) ? offgraphics : graph
+					.getGraphics();
+			if (index == -1)
+				return;
+			if (offgraphics != null || !graph.isXorEnabled()) {
+				dirty = graph
+						.toScreen((Rectangle2D) vertex.getBounds().clone());
+				Rectangle2D t = graph.toScreen(AbstractCellView
+						.getBounds(contextViews));
+				if (t != null)
+					dirty.add(t);
+			}
+			Rectangle2D newBounds = computeBounds(event);
+			if (graph.isXorEnabled()) {
+				g.setColor(graph.getForeground());
+				g.setXORMode(graph.getBackground().darker());
+				overlay(g);
+			} else {
+				firstOverlayInvocation = false;
+			}
+			if (cachedBounds != null)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It seems unusual to set the value of a variable only when it is already set. You might want to switch the null check of != on line 370 to == check before setting the field value on line 371 or verify if the check is needed at all.